lib/model.js

```diff
@@ -299,6 +299,10 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
       }
     }
 
+    if (ignoresMatchedDefault(properties[p]) && properties[p].default === propVal) {
```

Comment:
Nitpick: The value comparison here works for primitive types like number/string/etc...but I am not sure would it also return true for data in complicated types, like Date or other models.

Reply:
Looks good Yaapa, but I agree with @jannyHou 's question. Is there recursive code that handles nested object properties so that all properties are investigated?

Reply:
fair enough 👍